unity-ggjj/Assets/Scripts/TextDecoder/NarrativeScriptPlaylist.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+[RequireComponent(typeof(SceneLoader))]
+public class NarrativeScriptPlaylist : MonoBehaviour
+{
+    [field: Tooltip("List of narrative scripts to be played in order")]
+    [field: SerializeField] public List<NarrativeScript> NarrativeScripts { get; private set; }
+    
+    [field: Tooltip("List of narrative scripts to be used when taking penalties")]
+    [field: SerializeField] public List<NarrativeScript> FailureScripts { get; private set; }
+    
+    [field: Tooltip("A narrative script to be played on game over")]
+    [field: SerializeField] public NarrativeScript GameOverScript { get; private set; }
+    
+    [Tooltip("Write the name of the next scene to load here.")]
+    [SerializeField] private string _nextSceneName;
+
+    private SceneLoader _sceneLoader;
+    private int _narrativeScriptIndex = -1;
+
+    public NarrativeScript NarrativeScript => NarrativeScripts[_narrativeScriptIndex];
+
+    /// <summary>
+    /// Initializes variables
+    /// </summary>
+    private void Awake()
+    {
+        _sceneLoader = GetComponent<SceneLoader>();
+        InitializeNarrativeScripts(NarrativeScripts);
+        InitializeNarrativeScripts(FailureScripts);
+        GameOverScript.Initialize();
+    }
+
+    /// <summary>
+    /// Loops through a collection of narrative scripts and initialises each one
```

Comment:
Nit: Inconsistency: `Initialises` -> `Initializes`
```suggestion
    /// Loops through a collection of narrative scripts and initializes each one
```